src/components/CreateContract/CreateContractForms/team/container.js

```diff
@@ -1,54 +1,37 @@
 import React, { Component } from 'react';
 import { PropTypes as PT } from 'prop-types';
-import { getAllClients } from '../../../../services/clientService';
-import { getTeamsFromClient } from '../../../../services/teamService';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
 
-export default Wrapped =>
+import { updateSelectedTeamId } from '../../../../store/actions/teams';
+import { updateSelectedClientId } from '../../../../store/actions/clients';
+import { getClientOptions, getSelectedClientId, getTeamsOptions } from '../../../../store/reducers';
+
+
+const ContractTeamFormContainer = Wrapped =>
   class extends Component {
     static propTypes = {
+      updateSelectedClientId: PT.func,
+      updateSelectedTeamId: PT.func,
+      clientOptions: PT.array,
+      teamOptions: PT.array,
       onSuccess: PT.func,
     };
 
     constructor(props) {
       super(props);
       this.state = {
-        clients: [],
-        teams: [],
         teamResults: '',
       };
     }
 
-    componentDidMount() {
-      this.getClients();
-    }
-
-    getClients = () => {
-      getAllClients().then(response => {
-        const clients = response.data;
-        const clientsFormatted = clients.reduce((acc, client) => {
-          acc.push({
-            value: client.clientId,
-            displayValue: client.clientName,
-          });
-          return acc;
-        }, []);
-        clientsFormatted.unshift({
-          value: '-1',
-          displayValue: 'Please select a client',
-        });
-        this.setState({
-          clients: clientsFormatted,
-        });
-      });
-    };
+    handleTeamSectionSubmit = ({ selectedClientId, selectedTeamId }) => {
 
-    handleTeamSectionSubmit = ({selectedClientId, selectedTeamId}) => {
-
-      let selectedClient = this.state.clients.filter(
+      let selectedClient = this.props.clientOptions.filter(
         user => user.value == selectedClientId
       )[0];
 
-      let selectedTeam = this.state.teams.filter(
+      let selectedTeam = this.props.teamOptions.filter(
         user => user.value == selectedTeamId
       )[0];
```

Comment:
Similar to the other PR, `Array.find()` is a better way to find a single element in an array.

Reply:
Wasn't aware of find, but really like it. cheers. this has been updated.